concent_api/core/views.py

```diff
@@ -326,6 +328,31 @@ def receive_out_of_band(request, _message):
     return None
 
 
+@require_GET
+def protocol_constants(_request):
+    """ Endpoint which returns Concent time settings. """
+    return JsonResponse(
+        data = {
+            'concent_messaging_time': {
+                'name':  'CONCENT_MESSAGING_TIME',
+                'value': getattr(settings, 'CONCENT_MESSAGING_TIME', 'undefined')
```

Comment:
Please don't add default values here. We want this code to fail as soon as possible and as loudly as possible if these settings aren't defined so that we can fix this. The only case where they're undefined is if someone has completely botched his settings.

Might be a good idea to add a Django check for these settings too. And set them all to zeros in `testing.py` settings so that all tests that rely on them have to explicitly override them.

Reply:
:+1: But I don't think adding them to Django checks is necessary. When you look at `system_check.py` code you will see that a most of it looks redundant. If we want to do this lets create a separate issue and do it more generic way.

Reply:
Well, I'm OK with putting it in a separate issue. Could you create it?

It isn't really all that redundant. We don't have these checks anywhere else. We just write code as if this situation could never happen. At most we might have assertions in cases where we know the code would not just crash on its own but these don't count as checks. They're for programmers and run only in debug mode.

Checks are for the admin who sets up the application (that's still us for now ;)). Letting it crash when the setting is needed is much better than ignoring the problem but it's even better to add those checks so that bad settings fail immediately at `manage.py check` (which we always run before deployment) rather than long after the server has been deployed and just happens to enter the function that uses the setting that's missing. We did have deployment problems in the past caused by missing or bad settings. We need more of those checks because they're useful.

Reply:
:+1: Yes, I will add it tomorrow because it will need some more thinking what else can we check there.